### PR TITLE
Added German Translation

### PR DESCRIPTION
**Pull request by matthias-christen**
- Added German translation + tests
- I noticed that in the French and Spanish translations the "BE" (= Buddhist Era) (after the year) in English was misinterpreted to mean "BC" (= Before Christ). Added and corrected support for "BC" / "AD" in English, French, Spanish, German.
- Added support to parse "tomorrow morning", "yesterday evening", ... in English.

### CODE REVIEW

**Comment by Schultzer** [comment]
You should follow the changes in DateSlashFormatParser in PR #142

File: src/parsers/DE/DESlashDateFormatParser.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+var ParsedResult = require('../../result').ParsedResult;
+
+var PATTERN = new RegExp('(\\W|^)' +
+    '(?:' +
+        '(?:am\\s*?)?' +
+        '((?:sonntag|so|montag|mo|dienstag|di|mittwoch|mi|donnerstag|do|freitag|fr|samstag|sa))' +
+        '\\s*\\,?\\s*' +
+        '(?:den\\s*)?' +
+    ')?' +
+    '([0-9]{1,2})[\\/\\.\\-]([0-9]{1,2})' +
```
Comment (Schultzer) [inline src/parsers/DE/DESlashDateFormatParser.js]: There is a bug here #143 there is a bugfix for it in PR #142 you can take a look there for the fix.

File: src/parsers/DE/DESlashDateFormatParser.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    '(?:' +
+        '[\\/\\.\\-]' +
+        '([0-9]{4}|[0-9]{2})' +
```
Comment (Schultzer) [inline src/parsers/DE/DESlashDateFormatParser.js]: There is a bug here #143 there is a bugfix for it in PR #142

**Comment by matthias-christen** [comment]
As per suggestion of @Schultzer, this PR was split into 3 smaller PRs. Closing this one.

**Pull request closed without merging**